### PR TITLE
feat(server identity): Attempt to connect to local server on load

### PR DESCRIPTION
**Pull request by Camc314**
Checks if localhost:8096 is a valid server, when on the select server page

**Edit**
Attempts to connect to server specified in `config.json`

### CODE REVIEW

**Comment by bbogdanov** [comment]
I see this one to be addressed with PR for #139.

**Comment by Camc314** [comment]
Ah I though #139 was for connecting to a server specified in a config.json file?

**Comment by bbogdanov** [comment]
> Ah I though #139 was for connecting to a server specified in a config.json file?

Yes, but not only. It addresses

- config.json
- provided ones from the UI (stored in the localStorage)
- localhost - auto select if nothing from the list above is provided or if the last selected in previous session is localhost

File: store/servers.ts [hunk 1]
```diff
@@ -56,16 +68,15 @@ export const actions: ActionTree<ServerState, ServerState> = {
               address: serverUrl
             });
           }
-
-          this.$router.push('/login');
+          if (!silent) this.$router.push('/login');
         }
-      } else {
+      } else if (!silent) {
```
Comment (bbogdanov) [inline store/servers.ts:73]: This one will open a snackBar with that `server version is low` even if it is not.

File: store/servers.ts [hunk 2]
```diff
@@ -56,16 +68,15 @@ export const actions: ActionTree<ServerState, ServerState> = {
...
         dispatch('notifyServerVersionIsLow');
       }
     } catch (err) {
       // eslint-disable-next-line no-console
       console.error(err); // in case something inside the try rather than a request failure
-      dispatch('notifyServerCantBeFound');
+      if (!silent) dispatch('notifyServerCantBeFound');
```
Comment (bbogdanov) [inline store/servers.ts:79]: This one is inspired from spending significant amount of time to understand that at some point I got `server not found` error. Bringing back a `silent` variation is just getting back to the situation something spending a lot of time to understand some code suppressed an important message.

Comment (bbogdanov) [reply store/servers.ts:79]: Maybe if you do a middleware to listen for that action and if it was fired then execute another one to select localhost.

Comment (Camc314) [reply store/servers.ts:79]: The reason the error is supressed, is that the user might not want to connect to `localhost:8096`, or a server might not exist there, we are only checking it as a matter of convenience. If they attempt to type in the address manually, and connect the error will still show.

**Comment by ferferga** [comment]
@Camc314 Why localhost and not 127.0.0.1?

**Comment by Camc314** [comment]
> @Camc314 Why localhost and not 127.0.0.1?
> 
> We can avoid the dns query and I'm not sure how localhost will behave inside Docker (with server + Vue)

No reason. I'll update it

File: store/servers.ts (hunk 2 above)
Comment (bbogdanov) [reply store/servers.ts:79]: If the user might not want to connect, why do we force it in the first place?

Comment (Camc314) [reply store/servers.ts:79]: Purely as a check for convenience. In the future, the Address that is checked will be specified in a config.Json file

**Comment by MrTimscampi** [comment]
I think this needs to only connect automatically if in single server mode. Otherwise, it should just add the local server to the list of known servers, but not connect to it automatically. So it likely needs the config.json PR to be done, so we have the multi-server settings present.

**Pull request closed without merging**